pkg/deploy/controller/deployment/controller.go

```diff
@@ -223,6 +232,7 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 				},
 			},
 			ActiveDeadlineSeconds: &maxDeploymentDurationSeconds,
+			NodeSelector:          nodeSelector,
```

Comment:
That's true, the present code is an artifact of me looking at TemplateRef in addition to Template for the RC and hence constructing the nodeSelector outside. However, TemplateRef is not going to make it in for 1.0 and hence it doesn't matter for now.